src/root_finding/automatic_differentiation.jl

```diff
@@ -40,6 +40,7 @@ end
 
 sin(x::Jet) = Jet(sin(x.val), x.der*cos(x.val))
 cos(x::Jet) = Jet(cos(x.val), -x.der*sin(x.val))
+tan(x::Jet) = Jet(tan(x.val), x.der / cos(x.val)^2 )
```

Comment:
Since `sec(x)` is implemented directly as `one(T)/cos(x)`, I don't think that anything is gained by this (and in fact it could even be slightly less efficient).